can.c

```diff
@@ -34,29 +33,59 @@
 
 #include "microcom.h"
 
-enum socket_local_enum {
-	SOCKET_CANSIDE,
-	SOCKET_TERMSIDE,
-	SOCKET_MAX,
-};
-
 struct can_data {
-	int socket_can;
-	int socket_local[SOCKET_MAX];
 	int can_id;
 };
 
 static struct can_data data;
-static pthread_t can_thread;
 
 static ssize_t can_write(struct ios_ops *ios, const void *buf, size_t count)
 {
-	return write(ios->fd, buf, count);
+	size_t loopcount;
+	ssize_t ret = 0, err;
+
+	struct can_frame to_can = {
+		.can_id = data.can_id,
+	};
+
+	while (count > 0) {
+		loopcount = min(count, sizeof(to_can.data));
+		memcpy(to_can.data, buf, loopcount);
+		to_can.can_dlc = loopcount;
+retry:
+		err = write(ios->fd, &to_can, sizeof(to_can));
+		if (err < 0 && errno == EINTR)
+			goto retry;
+
+		if (err < 0)
+			return err;
+
+		assert(err < count);
```

Comment:
`ret` is used as return code value in this function and has a value that must be preserved. How would you restructure to make this nicer in your eyes?

Reply:
Didn't notice the other `ret`. My preference would be for `ret` to be `total` or `nbytes` and `err` to be `ret`.

Reply:
Probably a question of taste ... I like `ret` to be used to prepare the function's return value. I guess the code base isn't uniform here. Some policy on this would be good, but this is something I'd like to handle separately.